### DISCUSSION TITLE
Release of ebpf-for-windows binaries

### DISCUSSION THREAD

**Discussion by song-jiang** [microsoft/ebpf-for-windows#1188] [Announcements]
I'm wondering when we are going to have `ebpf-for-windows` release in a regular basis. The reason I ask is that I found it adds complexity if I have to build the binaries myself every time I need to test a recent code base. 

I was testing an old binaries built by me a few month ago. I'm hitting an issue `invalid argument` when creating a lpm_trie map. I'd like to see if latest code fixed the issue and failed to build latest code. Issue: https://github.com/microsoft/ebpf-for-windows/issues/1181

I can grab the latest release and raise issue against it if we have a regular release. I don't need to build the binaries myself in this case.

**Comment by dthaler** [comment]
This is covered by issue #1149 which is currently targeted for the current milestone.

**Reply by song-jiang** [reply]
Nice! @dthaler , could you share the link to download the latest "nuget" release?

**Reply by dthaler** [reply]
https://www.nuget.org/packages/eBPF-for-Windows/
This is the dev SDK not the runtime binaries

**Comment by Alan-Jowett** [comment]
Note:
You can also grab the artifacts from our [schedules CI/CD build](https://github.com/microsoft/ebpf-for-windows/actions/workflows/cicd.yml?query=event%3Aschedule++). The build produces the full set of binaries as well as the nuget package and the .MSI (note the MSI currently only places the binaries, doesn't install drivers or create the service).

**Comment by dthaler** [comment]
We now have a roughly monthly pre-release (i.e., unsigned binaries) after the end of each milestone, having done 0.2.0 after the 2206 milestone, and 0.3.0 after the 2207 milestone.   They can be downloaded from https://github.com/microsoft/ebpf-for-windows/releases

we also update the nuget package at the same time at https://www.nuget.org/packages/eBPF-for-Windows so to develop ebpf programs and apps for a given binary release, use the nuget package with the matching version number.

**Reply by dthaler** [reply]
Just did a pre-release of 0.4.0, with changelog visible at https://github.com/microsoft/ebpf-for-windows/releases.

**Reply by dthaler** [reply]
Version 0.5.0 is now available, with changelog visible at Just did a pre-release of 0.4.0, with changelog visible at https://github.com/microsoft/ebpf-for-windows/releases.
